rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -2614,6 +2614,164 @@ TEST_F(TestNode, declare_parameter_with_vector) {
   }
 }
 
+// test non-array data types for declare parameter function templates that are explicitly defined
+TEST_F(TestNode, declare_parameter_allowed_simple_types_function_templates) {
+  auto node = std::make_shared<rclcpp::Node>(
+    "test_declare_parameter_allowed_simple_types_function_templates"_unq);
+  {
+    // declare parameter and then get types to check
+    auto name1 = "parameter"_unq;
+    auto name2 = "parameter"_unq;
+    auto name3 = "parameter"_unq;
+    auto name4 = "parameter"_unq;
+    auto name5 = "parameter"_unq;
+    auto name6 = "parameter"_unq;
+    auto name7 = "parameter"_unq;
+
+    node->declare_parameter<bool>(name1, false);
+    node->declare_parameter<int>(name2, 1234);
+    node->declare_parameter<int64_t>(name3, 12340);
+    node->declare_parameter<float>(name4, static_cast<float>(12.34));
```

Comment:
Nit: here, and elsewhere, you can just do:

```suggestion
    node->declare_parameter<float>(name4, 12.34f);
```

(but it ends up being the same, so I won't block on it)